kyaml/setters2/settersutil/substitutioncreator.go

```
@@ -62,3 +73,106 @@ func (c SubstitutionCreator) Create(openAPIPath, resourcesPath string) error {
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 }
+
+// CreateSettersForSubstitution creates the setters for all the references in the substitution
+// values if they don't already exist in openAPIPath file.
+func (c SubstitutionCreator) CreateSettersForSubstitution(openAPIPath string) error {
+	b, err := ioutil.ReadFile(openAPIPath)
+	if err != nil {
+		return err
+	}
+
+	// parse the yaml file
+	y, err := yaml.Parse(string(b))
+	if err != nil {
+		return err
+	}
+
+	m, err := c.GetValuesForMarkers()
+	if err != nil {
+		return err
+	}
+
+	// for each ref in values, check if the setter already exists, if not create them
+	for _, value := range c.Values {
+		obj, err := y.Pipe(yaml.Lookup(
+			// get the setter key from ref. Ex: from #/definitions/io.k8s.cli.setters.image_setter
+			// extract io.k8s.cli.setters.image_setter
+			"openAPI", "definitions", strings.Split(value.Ref, "/")[2]))
+
+		if err != nil {
+			return err
+		}
+
+		if obj == nil {
+			sd := setters2.SetterDefinition{
+				// get the setter name from ref. Ex: from #/definitions/io.k8s.cli.setters.image_setter
+				// extract image_setter
+				Name:        strings.Split(value.Ref, ".")[4],
+				Value:       m[value.Marker],
+				SetBy:       "",
+				Description: "Created for substitution " + c.Name,
+				Count:       0,
+			}
+			sd.AddToFile(openAPIPath)
+		}
+	}
+	return nil
+}
+
+// GetValuesForMarkers parses the pattern and field value to derive values for the
+// markers in the pattern string. Returns error if the marker values can't be derived
+func (c SubstitutionCreator) GetValuesForMarkers() (map[string]string, error) {
+	m := make(map[string]string)
+	inds, _ := c.GetStartIndices()
```

is ignoring the error the right thing to do here?